dataset/src/main/scala/frameless/RecordEncoder.scala

```diff
@@ -44,10 +46,41 @@ object RecordEncoderFields {
     }
 }
 
-class RecordEncoder[F, G <: HList]
+trait NewInstanceExprs[T <: HList] {
```

Comment:
the failing tests suggest that this should extend `Serializable`

Reply:
fixed, hopefully :)